From bdcdefc402df6b9d9363c16d0209f9af7244b8b3 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:39:40 +0800
Subject: [PATCH 09/12] GSI Android 17 compatibility:
 0009-Allow-sys-vm-watermark_scale_factor-conflict-seen-on.patch

Change-Id: Ia83c2150494b7ecc9196d11e217294213cf7182b
---
 libsepol/cil/src/cil_post.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/libsepol/cil/src/cil_post.c b/libsepol/cil/src/cil_post.c
index ecc704e..29f4559 100644
--- a/libsepol/cil/src/cil_post.c
+++ b/libsepol/cil/src/cil_post.c
@@ -509,6 +509,15 @@ static int cil_post_genfscon_context_compare(const void *a, const void *b)
 		 */
 		if(strcmp(a_genfscon->path_str, "/devices/virtual/block/") == 0)
 			bypass = 1;
+        /*
+         * This conflict has been seen on Freebox Player Pop
+         * - AOSP T says (genfscon proc "/sys/vm/watermark_scale_factor" (u object_r proc_watermark_scale_factor ((s0) (s0))))
+         * - stock rom says proc_vm_writable
+         *
+         * Stock ROM uses it only in recovery so it's safe to ignore
+         */
+		if(strcmp(a_genfscon->path_str, "/sys/vm/watermark_scale_factor") == 0)
+			bypass = 1;
 		if(strcmp(a_genfscon->fs_str, "exfat") == 0 || strcmp(a_genfscon->fs_str, "esdfs") == 0) {
             if(strcmp(a_genfscon->path_str, "/") == 0)
                 bypass = 1;
